### PR TITLE
Slombard/cha 573 help with 468 user come back after sell

### PR DESCRIPTION
**Pull request by 552020**
Building on CHA-468

I moved the fetching function for add-user-to-agreed to it's own file like for send-message, and I adapted the resolver so that it reads what the new fetching functions returns and give 50/50 change for the chat to be sold or stay pending. I changed a little bit the logic to update the user, the user is now always added to the user_agreed array so the table are updated correctly. To me it seems to work.

This regards only 1 half of the issue, my Invitations. 

Errors are not handled and we should start to handle them, everytime we touch an endpoint.

### CODE REVIEW

**Comment by lmangall** [comment]
Hey @552020 I am also looking into @Danielg1406 's PR. Do you have a plan of what you want to do ?
PS: it's behind main (no conflicts)

**Comment by 552020** [comment]
> Hey @552020 I am also looking into @Danielg1406 's PR. Do you have a plan of what you want to do ? PS: it's behind main (no conflicts)

Yes, I do. I wrote also in Discord about my plan.

**Comment by lmangall** [comment]
I thought I answered but my message didn't got sent. 
If you are also working on this I'll do something else (like finish with the errors) so we don't risk conflicts or double work

**Comment by 552020** [comment]
I'm converting it back to draft @Danielg1406 because I'm still working on it.

**Comment by Danielg1406** [comment]
@552020 I can't believe you change it from 3 seconds to 2 😂

**Comment by lmangall** [comment]
> Errors are not handled and we should start to handle them, everytime we touch an endpoint.

It's almost done:
https://github.com/TON-42/new_vite_frontend/pull/63

File: src/mocks/messageHandlers.ts
Comment (lmangall) [inline src/mocks/messageHandlers.ts:1]: I think this file was replaced by sendMessageResolver.ts and we should get rid of it, but maybe I am missing something.

**Review by lmangall — APPROVED** [review]
LGTM

**Pull request merged**